test/Test/Data/Array/NonEmpty.purs

```diff
@@ -297,12 +297,22 @@ testNonEmptyArray = do
   log "Unfoldable instance"
   assert $ U1.range 0 9 == NEA.range 0 9
 
-  log "foldl should work"
+  log "foldMap1 should work"
+  assert $ foldMap1 Additive (fromArray [1, 2, 3, 4]) == Additive 10
+
+  log "fold1 should work"
   -- test through sum
   assert $ sum (fromArray [1, 2, 3, 4]) == 10
 
-  log "foldMap1 should work"
-  assert $ foldMap1 Additive (fromArray [1, 2, 3, 4]) == Additive 10
+  log "foldr1 should work"
+  assert $ foldr1 (\l r -> "(" <> l <> r <> ")") (fromArray ["a", "b", "c", "d"]) == "(((ab)c)d)"
+  assert $ foldr1 (\l r -> "(" <> l <> r <> ")") (fromArray ["a", "b"])           == "(ab)"
+  assert $ foldr1 (\l r -> "(" <> l <> r <> ")") (fromArray ["a"])                == "a"
+
+  log "foldl1 should work"
+  assert $ foldl1 (\l r -> "(" <> l <> r <> ")") (fromArray ["a", "b", "c", "d"]) == "(a(b(cd)))"
```

Comment:
Pretty sure this should be `foldr1`'s test:
```
"( <> "a" <>
  "(" <> "b" <>
    "(" <> "c" <>
      "(" <> "d" <>
      ")"
    ")"
  ")"
")"
```

Reply:
indeed, tnx, but the innermost d is not wrapped

the test is taken from https://github.com/purescript/purescript-nonempty/pull/39/files

Reply:
Thanks! I updated my code.